docs/sources/flow/setup/deploy-agent.md

```diff
@@ -13,3 +13,58 @@ weight: 900
 
 {{< docs/shared source="agent" lookup="/deploy-agent.md" version="<AGENT_VERSION>" >}}
 
+## Processing different types of telemetry in different Agent instances
+
+If the load on the Agents is small, it is recommended to process all necessary telemetry signals in the same Agent process. 
+For example, a single Agent can process all of the incoming metrics, logs, traces, and profiles.
+
+However, if the load on the Agents is big, it may be beneficial to process different telemetry signals in different deployments of Agents:
+* This provides better stability due to the isolation between processes.
+  * For example, an overloaded Agent processing traces won't impact an Agent processing metrics.
+* Different types of signal collection require different methods for scaling:
+  * "Pull" components such as `prometheus.scrape` and `pyroscope.scrape` are scaled using hashmod sharing or clustering. 
+  * "Push" components such as `otelcol.receiver.otlp` are scaled by placing a load balancer in front of them.
+
+### Traces
+
+Scaling Agent instances for tracing is very similar to [scaling OpenTelemetry Collector][scaling-collector] instances.
+This is because most Flow components used for tracing are based on components from the Collector.
+
+[scaling-collector]: https://opentelemetry.io/docs/collector/scaling/
+
+#### When to scale
+
+To decide whether scaling is necessary, check metrics such as:
+* `receiver_refused_spans_ratio_total` from receivers such as `otelcol.receiver.otlp`.
+* `processor_refused_spans_ratio_total` from processors such as `otelcol.processor.batch`.
+* `exporter_send_failed_spans_ratio_total` from exporters such as `otelcol.exporter.otlp` and `otelcol.exporter.loadbalancing`.
+
+#### Stateful and stateless components
+
+In the context of tracing, a "stateful component" is a component 
+which needs to aggregate certain spans in order to work correctly.
+A "stateless Agent" is an Agent which does not contain stateful components.
+
+Scaling stateful Agents is more difficult, because spans must be forwarded to a 
+specific Agent according to a span property such as trace ID or a `service.name` attribute.
+This can be done using `otelcol.exporter.loadbalancing`.
+
+Examples of stateful components:
+
+* `otelcol.processor.tail_sampling`
+* `otelcol.connector.spanmetrics`
+* `otelcol.connector.servicegraph`
+
+<!-- TODO: link to the otelcol.exporter.loadbalancing docs for more info -->
+
+A "stateless component" does not need to aggregate specific spans in order to work correctly - 
+it can work correctly even if it only has some of the spans of a trace.
```

Comment:
```suggestion
A "stateless component" does not need to aggregate specific spans to work correctly - 
it can work correctly even if it only has some of the spans of a trace.
```